go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -96,100 +93,80 @@ func newEgressFirewallRule(rawEgressFirewallRule egressfirewallapi.EgressFirewal
-							logicalSwitch,
-							"acls",
-							egressFirewallACLID,
-						)
-						if err != nil {
-							klog.Errorf("Unable to remove egress firewall acl: %s on %s, cannot cleanup old stale data, stderr: %s, err: %v", egressFirewallACLID, logicalSwitch, stderr, err)
+
+		// Mode is shared gateway mode, make sure to delete all egfw ACLs on the node switches
+		if len(egressFirewallACLs) != 0 {
+			for _, egressFirewallACL := range egressFirewallACLs {
+
+				nodeLogicalSwitchLookup := func(item *nbdb.LogicalSwitch) bool {
+					// Ignore external and Join switches
+					if strings.Contains(item.Name, "join") || strings.Contains(item.Name, "ext") {
+						return false
+					}
+
+					for _, acl := range item.ACLs {
+						if acl == egressFirewallACL.UUID {
+							return true
 						}
 					}
+					return false
 				}
+
+				nodeSwitches, err := libovsdbops.FindSwitch(oc.modelClient.GetClient(), nodeLogicalSwitchLookup)
+				if err != nil {
+					klog.Errorf("Error while querying ACLs uuid:%s with reject action: %s", egressFirewallACL.UUID, err)
+					return
+				}
+
+				err = libovsdbops.RemoveACLFromSwitches(oc.modelClient, nodeSwitches, egressFirewallACL.UUID)
+				if err != nil {
+					klog.Errorf("Failed to remove reject acl from logical switches: %v", err)
+					return
+				}
+
 			}
 		}
 	}
-	egressFirewallACLIDs, stderr, err := util.RunOVNNbctl(
-		"--data=bare",
-		"--no-heading",
-		"--columns=_uuid",
-		"--format=table",
-		"find",
-		"acl",
-		fmt.Sprintf("priority<=%s", types.EgressFirewallStartPriority),
-		fmt.Sprintf("priority>=%s", types.MinimumReservedEgressFirewallPriority),
-		fmt.Sprintf("direction=%s", types.DirectionFromLPort),
-	)
-	if err != nil {
-		klog.Errorf("Unable to list egress firewall logical router policies, cannot convert old ACL data, stderr: %s, err: %v", stderr, err)
-		return
-	}
-	if egressFirewallACLIDs != "" {
-		for _, egressFirewallACLID := range strings.Fields(egressFirewallACLIDs) {
-			_, stderr, err := util.RunOVNNbctl(
-				"set",
-				"acl",
-				egressFirewallACLID,
-				fmt.Sprintf("direction=%s", types.DirectionToLPort),
-			)
-			if err != nil {
-				klog.Errorf("Unable to set ACL direction on egress firewall acl: %s, cannot convert old ACL data, stderr: %s, err: %v", egressFirewallACLID, stderr, err)
-			}
+	// update the direction of each egressFirewallACL if needed
+	if len(egressFirewallACLs) != 0 {
+		opModels := []libovsdbops.OperationModel{}
+		for i, egressFirewallACL := range egressFirewallACLs {
+			aclName := egressFirewallACLs[i].Name
+			egressFirewallACL.Direction = types.DirectionToLPort
+			opModels = append(opModels, libovsdbops.OperationModel{
+				Model:          egressFirewallACL,
+				ModelPredicate: func(acl *nbdb.ACL) bool { return acl.Name == aclName },
+				OnModelUpdates: []interface{}{
+					&egressFirewallACL.Direction,
+				},
+				ErrNotFound: true,
+				BulkOp:      true,
+			})
+		}
+		if _, err := oc.modelClient.CreateOrUpdate(opModels...); err != nil {
```

Comment:
Why not use the existing `libovsdbops.CreateOrUpdateACLs` (and optionally change the implementation to use modelClient internally)?

Reply:
That's on the list, however I think I would like to tackle this in a follow up PR if that's alright